prog.go

```diff
@@ -1000,3 +1013,41 @@ func findTargetInProgram(prog *Program, name string, progType ProgramType, attac
 
 	return spec.TypeID(targetFunc)
 }
+
+// find a kfunc in a kernel module or vmlinux
+func findKfuncInKernel(kernelSpec *btf.Spec, kfunc *btf.Func) (*btf.Handle, btf.TypeID, error) {
+	var fn *btf.Func
+	err := kernelSpec.TypeByName(kfunc.Name, &fn)
+	if errors.Is(err, btf.ErrNotFound) {
+		// couldn't find kfunc in vmlinux, searching modules.
+		module, id, err := findTargetInModule(kernelSpec, kfunc.Name, kfunc)
+		if errors.Is(err, btf.ErrNotFound) {
+			return nil, 0, fmt.Errorf("kfunc '%s' doesn't exist in vmlinux or modules: %v: %w", kfunc.Name, err, btf.ErrNotFound)
+		}
+
+		if err != nil {
+			return nil, 0, fmt.Errorf("couldn't find kfunc '%s' in modules: %v: %w", kfunc.Name, err, ErrNotSupported)
+		}
+
+		if err := btf.CheckTypeCompatibility(kfunc.Type, kfunc.Type); err != nil {
```

Comment:
Would it be possible to invoke this in `fixupKfuncs` instead? That would reduce some duplication.

Reply:
Fixed :ballot_box_with_check: 
how about this?

Reply:
Now the `btf.CheckTypeCompatibility` checks are wrong IMO. Now we call it in `findKfuncInKernel` and `fixupKfuncs`, and in `fixupKfuncs` we check `btf.CheckTypeCompatibility(kfm.Type, kfm.Type)` which will always pass.

Reply:
@rgo3, 
I'm sorry for the late reply, and for made a degrade. 
I have read your changes, thank you.

Reply:
I'm still not super happy with what I added, but that way we only do the type compatibility check once. Let's discuss next week with Lorenz, we should be close to the finish line :muscle:

Reply:
🙇

Reply:
I had to do some additional refactoring, but I think it ended up fine / not much more complex. PTAL @rgo3